addons/payment/models/payment_transaction.py

```diff
@@ -180,7 +180,7 @@ def create(self, values_list):
             # Duplicate partner values
             partner = self.env['res.partner'].browse(values['partner_id'])
             values.update({
-                'partner_name': partner.name,
+                'partner_name': partner.name or partner.parent_id.name or "unknown", # fallback in case of invoice address without name,
```

Comment:
I think that is enough to do
```python
'partner_name': partner.name or partner.parent_id.name
```

`partner` in the case of a sale_order is the field `partner_invoice_id` and its parent is the `partner_id` of the sale order that has always a name.
Although this should be done in sales and not in payment if possible, as the invoice partner is a feature of sales.

Reply:
I agree with the fact that the third option can probably be removed.

About the fact that it is coming from Sales, I don't because the difficulty is that there is only one `partner_id` assigned to the transaction, and this one is not chosen based on the fact that the field `name` is empty or not. So it will always be the `partner_invoice_id` in Sales routes. So I think it makes more sense to me to handle it here.